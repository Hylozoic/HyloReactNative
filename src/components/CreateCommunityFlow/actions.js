export const MODULE_NAME = `CreateCommunityFlow`
// export const ADD_COMMUNITY_NAME = `${MODULE_NAME}/ADD_COMMUNITY_NAME`
// export const ADD_COMMUNITY_DOMAIN = `${MODULE_NAME}/ADD_COMMUNITY_DOMAIN`
// export const ADD_COMMUNITY_PRIVACY = `${MODULE_NAME}/ADD_COMMUNITY_PRIVACY`

export const FETCH_COMMUNITY_EXISTS = `${MODULE_NAME}/FETCH_URL_EXISTS`

export function fetchCommunityExists (slug) {
  return {
    type: FETCH_COMMUNITY_EXISTS,
    graphql: {
      query: `
        query ($slug: String) {
          communityExists (slug: $slug) {
            exists
          }
        }
      `,
      variables: {
        slug
      }
    }
  }
}
