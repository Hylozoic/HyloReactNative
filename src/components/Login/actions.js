import { clearSessionCookie } from '../../util/session'
import { LoginManager } from 'react-native-fbsdk'
import { GoogleSignin } from '@react-native-community/google-signin'
import { isEmpty } from 'lodash'

export const LOGIN = 'LOGIN'
export const LOGIN_BY_TOKEN = 'LOGIN_BY_TOKEN'
export const LOGIN_WITH_FACEBOOK = 'LOGIN_WITH_FACEBOOK'
export const LOGIN_WITH_GOOGLE = 'LOGIN_WITH_GOOGLE'
export const LOGOUT = 'LOGOUT'

export function login (email, password) {
  return {
    type: LOGIN,
    payload: {
      api: {method: 'post', path: '/noo/login', params: {email, password}}
    },
    meta: {email}
  }
}

export function loginByToken (userId, loginToken) {
  return {
    type: LOGIN_BY_TOKEN,
    payload: {
      api: {method: 'post', path: '/noo/login/token', params: {u: userId, t: loginToken}}
    }
  }
}

export function loginWithFacebook (accessToken) {
  return {
    type: LOGIN_WITH_FACEBOOK,
    payload: {
      api: {method: 'post', path: `/noo/login/facebook-token/oauth?access_token=${accessToken}`}
    }
  }
}

export function loginWithGoogle (accessToken) {
  return {
    type: LOGIN_WITH_GOOGLE,
    payload: {
      api: {method: 'post', path: `/noo/login/google-token/oauth?access_token=${accessToken}`}
    }
  }
}

export function logout () {
  return {
    type: LOGOUT,
    payload: {
      api: {
        method: 'delete',
        path: '/noo/session',
        transform: () =>
          clearSessionCookie()
          .then(() => LoginManager.logOut())
          .then(() => {
            if (!isEmpty(GoogleSignin.currentUser())) {
              return GoogleSignin.signOut()
            }
          })
      }
    }
  }
}
