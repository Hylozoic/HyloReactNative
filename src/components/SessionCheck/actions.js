import { getSessionCookie } from '../../util/session'
export const CHECK_SESSION = 'CHECK_SESSION'
export const CHECK_VERSION = 'CHECK_VERSION'

export function checkSession () {
  return {
    type: CHECK_SESSION,
    payload: getSessionCookie().then(cookie => {
      if (!cookie) return false

      return {
        api: {
          path: '/noo/user/status',
          transform: json => !!json.signedIn
        }
      }
    })
  }
}

export function checkVersion (platform, version) {
  const queryParams = `${platform}-version=${version}`
  return {
    type: CHECK_VERSION,
    payload: {
      api: {method: 'get', path: `/noo/mobile/auto-update-info?${queryParams}`}
    }
  }
}
