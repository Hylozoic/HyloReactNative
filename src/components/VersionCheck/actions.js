export const CHECK_VERSION = 'CHECK_VERSION'

export function checkVersion (platform, version) {
  const queryParams = `${platform}-version=${version}`
  return {
    type: CHECK_VERSION,
    payload: {
      api: {method: 'get', path: `/noo/mobile/check-should-update?${queryParams}`}
    }
  }
}
