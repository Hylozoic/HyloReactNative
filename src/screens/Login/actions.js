import { clearSessionCookie } from 'util/session'
import { LoginManager } from 'react-native-fbsdk'
import { GoogleSignin } from '@react-native-community/google-signin'
import { isEmpty } from 'lodash'

export const LOGIN = 'LOGIN'
export const LOGIN_BY_TOKEN = 'LOGIN_BY_TOKEN'
export const LOGIN_WITH_APPLE = 'LOGIN_WITH_APPLE'
export const LOGIN_WITH_FACEBOOK = 'LOGIN_WITH_FACEBOOK'
export const LOGIN_WITH_GOOGLE = 'LOGIN_WITH_GOOGLE'
export const LOGOUT = 'LOGOUT'

export function login (email, password) {
  return {
    type: LOGIN,
    payload: {
      api: { method: 'post', path: '/noo/login', params: { email, password } }
    },
    meta: { email }
  }
}

export function loginByToken (userId, loginToken) {
  return {
    type: LOGIN_BY_TOKEN,
    payload: {
      api: { method: 'post', path: '/noo/login/token', params: { u: userId, t: loginToken } }
    }
  }
}

export function loginWithApple (params) {
  return {
    type: LOGIN_WITH_APPLE,
    payload: {
      api: { method: 'post', path: '/noo/login/apple/oauth', params }
    }
  }
}

export function loginWithFacebook (accessToken) {
  return {
    type: LOGIN_WITH_FACEBOOK,
    payload: {
      api: { method: 'post', path: `/noo/login/facebook-token/oauth?access_token=${accessToken}` }
    }
  }
}

export function loginWithGoogle (accessToken) {
  return {
    type: LOGIN_WITH_GOOGLE,
    payload: {
      api: { method: 'post', path: `/noo/login/google-token/oauth?access_token=${accessToken}` }
    }
  }
}

export function logout () {
  return {
    type: LOGOUT,
    payload: {
      api: {
        method: 'delete',
        path: '/noo/session',
        transform: () =>
          clearSessionCookie()
            .then(() =>
              LoginManager.logOut()
            )
            // TODO: This should be already handled by the same code in login/actions/Logout
            .then(async () => {
              if (!isEmpty(await GoogleSignin.getCurrentUser())) {
                return GoogleSignin.signOut()
              }
            })
      }
    }
  }
}
