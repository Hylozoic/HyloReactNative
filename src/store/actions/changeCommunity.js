import { CHANGE_COMMUNITY } from 'store/constants'

export default function changeCommunity (id) {
  return {
    type: CHANGE_COMMUNITY,
    payload: id
  }
}
