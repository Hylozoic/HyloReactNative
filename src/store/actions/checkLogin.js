import { get } from 'lodash/fp'
import { CHECK_LOGIN } from 'store/constants'

const getRoot = get('me')

export default function checkLogin () {
  return {
    type: CHECK_LOGIN,
    graphql: {
      query: `
        query CheckLogin {
          me {
            id
            avatarUrl
            email
            emailValidated
            hasRegistered
            name
            settings {
              alreadySeenTour
              digestFrequency
              dmNotifications
              commentNotifications
              locale
              signupInProgress
              streamChildPosts
              streamViewMode
              streamSortBy
              streamPostType
            }
          }
        }
      `
    },
    meta: {
      extractModel: [
        {
          getRoot,
          modelName: 'Me'
        }
      ]
    }
  }
}
