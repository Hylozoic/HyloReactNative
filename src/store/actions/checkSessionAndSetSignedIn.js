import { CHECK_SESSION_AND_SET_SIGNED_IN } from '../constants'
import { getSessionCookie } from 'util/session'

export default async function checkSessionAndSetSignedIn () {
  const cookie = await getSessionCookie()
  const payload = cookie ? {
    api: {
      path: '/noo/user/status',
      transform: json => {
        console.log('!!!! json in checkSessionAndSetSignedIn:', json)
        return !!json.signedIn
      } 
    }
  } : false

  return {
    type: CHECK_SESSION_AND_SET_SIGNED_IN,
    payload
  }
}
