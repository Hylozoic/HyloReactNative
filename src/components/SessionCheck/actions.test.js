import { checkSession, checkVersion } from './actions'

describe('checkSession', () => {
  it('works with a cookie', () => {
    jest.mock('../../util/session', () => ({
      getSessionCookie: () => Promise.resolve('cookie')
    }))
    expect(checkSession()).toMatchSnapshot()
  })

  it('works without a cookie', () => {
    jest.mock('../../util/session', () => ({
      getSessionCookie: () => Promise.resolve(false)
    }))
    expect(checkSession()).toMatchSnapshot()
  })
})

describe('checkVersion', () => {
  it('matches the existing snapshot', () => {
    const platform = 'ios'
    const version = '1.0'
    expect(checkVersion()).toMatchSnapshot()
  })
})
