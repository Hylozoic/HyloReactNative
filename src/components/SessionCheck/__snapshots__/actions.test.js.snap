// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkSession works with a cookie 1`] = `
Object {
  "payload": Promise {
    "_40": 0,
    "_55": null,
    "_65": 0,
    "_72": null,
  },
  "type": "CHECK_SESSION",
}
`;

exports[`checkSession works without a cookie 1`] = `
Object {
  "payload": Promise {
    "_40": 0,
    "_55": null,
    "_65": 0,
    "_72": null,
  },
  "type": "CHECK_SESSION",
}
`;

exports[`checkVersion matches the existing snapshot 1`] = `
Object {
  "payload": Object {
    "api": Object {
      "method": "get",
      "path": "/noo/mobile/auto-update-info?undefined-version=undefined",
    },
  },
  "type": "CHECK_VERSION",
}
`;
